doc/MSTP/MSTP.md

```diff
@@ -0,0 +1,1161 @@
+
+## Del VLAN from  Exisiting Instance
+![MSTP VLAN Del](images/MSTP_Del_ExistingInstance.drawio.png)
+
+
+# Configuration Commands
+Following configuration commands will be provided for configuration of MSTP:
+## Global Level
+- **config spanning_tree {enable|disable} {mst}**
+  - Enables or disables mstp at global level on all ports of the switch.
+  - Only one mode of STP can be enabled at a time.
+  - By default disabled.
+    
+- **config spanning_tree max_hops \<max-hops-value\>**
+  - Specify the number of maximum hops before the BPDU is discarded inside a region.
+  - max-hops-value: Default: 20, range: 1-255
+
+- **config spanning_tree hello \<hello-value\>**
+  - Specify configuring hello interval in sec for transmission of BPDUs. 
+  - Default: 2, range 1-10 
+
+- **config spanning_tree max_age \<max_age-value\>**
+  - Specify configuring maximum time to listen for root bridge in seconds.
+  - Default: 2, range 6-40
+
+
+## Region Level
+Below commands allow configuring on region basis:
+
+- **config spanning_tree mst region-name \<region-name\>**
+  - Edit the name of region
+  - region-name: Case sensitive, characters should be less than or equal to 32, default: mac-address of bridge
+- **config spanning-tree mst revision \<revision-number\>**
+  - Revision number is used to track changes in the configuration and to synchronize the configuration across the switches in the same region.
+  - revision-number: Default: 0, range: 0-65535
+
+## Instance Level
+
+Below commands allow configuration of an instance:
+
+- **config spanning_tree mst instance \<instance-id\> priority \<priority-value\>** 
+  - Configure priority of bridge for an instance.
+  - instance-id: id of the instance for which bridge priority is to be defined. If the provided instance id is not created yet, an error message is displayed.
+  - priority-value: Default: 32768, range: 0-61440 (should be multiple of 4096)
+- **config spanning_tree mst instance \<instance-id\> vlan (add|del)  \<vlan-id\>**
+  - VLAN to instance mapping.
+  - instance-id: id of the instance to which VLAN is to be mapped. If the provided instance id is not created yet, an error message is displayed.
+  - vlan-id: Range: 1-4094. If the provided VLAN is not created yet, an error message is displayed.
+  - Instance is only active when there is at least one VLAN member port configured for one of the mapped VLANs.
+
+## Instance, Interface Level
+Following commands are used for spanning-tree configurations on per instance, per interface basis:
+
+- **config spanning_tree mst instance \<instance-id\> interface \<ifname\> priority \<priority-value\>**
+  - Configure priority of an interface for an instance.
+  - priority-value: Default: 128, range: 0-240
+- **config spanning_tree mst instance \<instance-id\> interface \<ifname\> cost \<cost-value\>**
+  - Configure path cost of an interface for an instance.
+  - cost-value: Range: 1-200000000
+
+## Interface Level
```

Comment:
We will be reusing the PVST commands to support MST configurations as well.

Reply:
Kindly add this as a note that the existing interface level PVST commands would be reused for MSTP as well since PVST and MSTP would be mutually exclusive on the same Vlan-Port combination.